lib/CL/devices/common_utils.c

```diff
@@ -265,6 +265,9 @@ pocl_cpu_init_common (cl_device_id device)
   device->llvm_cpu = OCL_KERNEL_TARGET_CPU;
   if (device->llvm_cpu == NULL)
     device->llvm_cpu = pocl_get_llvm_cpu_name ();
+
+  if (strstr (OCL_KERNEL_TARGET, "x86") != NULL)
+    device->run_safe_division_pass = CL_TRUE;
```

Comment:
Maybe 'safe' is not the best description for it? Better could be _noexcept_ or _notrap_ similar?

I think the non-trapping divisions are actually less safe ones in this sense.

Reply:
> non-trapping divisions are actually less safe ones in this sense.

Can you explain what you mean, how are they less safe ?

Reply:
Because I think div-by-zero is usually a application bug and this pass prevents the bug from causing exceptions. Yep, in OpenCL case there's no observable difference as exceptions are not allowed, but I cannot understand why it would be 'safer' either.

Reply:
It is safer in the sense that it has defined behavior for corner cases. IOW, you can rely on the behavior being the same, on all architectures. "noexcept" or "notrap" would imply that UB results in a trap, which isn't always true. But i'll change the name to something else.